partiql-plan/src/main/resources/partiql_plan.ion

```diff
@@ -197,6 +197,26 @@ rex::{
     err::{
       message: string,
     },
+
+    // Information that worth to have to help end customer debugging the plan.
+    debug::[
+      call::[
+        unresolved::{
+          identifier: identifier,
+          args: list::[rex],
+        },
+      ],
+      problem::[
+        missing::{
+          trace: '.rex.op',
+          problem: string
+        },
+        error::{
+          trace: '.rex.op',
+          problem: string
+        },
+      ]
+    ]
```

Comment:
Essentially: 

Consider the query `1 + t.f2` where t is `t : {f1: INT2}`

The design was to have a trace of missing: 

```
Query[tag=Plan-780ab923]
      └── Missing[problem=-1:-1:<unknown>: int4, missing is/are incompatible data types for the 'PLUS' operator., tag=Plan-39c9df92, static_type=missing]
         └── Dynamic[tag=Plan-7452fb19]
            ├── Lit[value=Int32ValueImpl(value=1, annotations=[]), tag=Plan-6063c078, static_type=int4]
            ├── Missing[problem=-1:-1:<unknown>: Expression always returns null or missing: caused by Path Navigation always returns MISSING, tag=Plan-3defad5b, static_type=missing]
            │  └── Symbol[key=f2, tag=Plan-ccbc991]
            │     └── Global[tag=Plan-e021a9f7, static_type=struct(f1: int2, [Open(value=false), UniqueAttrs(value=true), Ordered])]
            │        └── Ref[catalog=0, symbol=8, tag=Plan-cfc1a326]
            ├── Candidate[tag=Plan-7e3c9eab]
            │  └── Ref[catalog=0, symbol=0, tag=Plan-6263a65a]
            | ....... 
```

Breaking it down a little bit: 
```
Query[tag=Plan-780ab923]
      └── Missing[problem=-1:-1:<unknown>: int4, missing is/are incompatible data types for the 'PLUS' operator., tag=Plan-39c9df92, static_type=missing]
```

Top level missing -> because 1 + MISSING is an unresolved function. 
Note that the idea is to have 1 + MISSING report type mismatch error, NOT plus function propagate missing. 
This is because the `isMissingCall` flag is set on each FnSignature.  
During function resolution, we first get all the variants based on function name. We never prohibit the possibility to have say two variants of the same function name, but has different `isMissingCall` flag. 

Next: 
```
└── Missing[problem=-1:-1:<unknown>: int4, missing is/are incompatible data types for the 'PLUS' operator., tag=Plan-39c9df92, static_type=missing]
         └── Dynamic[tag=Plan-7452fb19]
```

The RexOpMissing should have put all the candidate function to help debugging. This is essentially to show the user what is the actually input type and all the possible legal static type. 

Next: 
```
└── Dynamic[tag=Plan-7452fb19]
            ├── Lit[value=Int32ValueImpl(value=1, annotations=[]), tag=Plan-6063c078, static_type=int4]
            ├── Missing[problem=-1:-1:<unknown>: Expression always returns null or missing: caused by Path Navigation always returns MISSING, tag=Plan-3defad5b, static_type=missing]
            │  └── Symbol[key=f2, tag=Plan-ccbc991]
            │     └── Global[tag=Plan-e021a9f7, static_type=struct(f1: int2, [Open(value=false), 
```
The interesting part is the second arg, which contains another rexOpMissing. The idea is to have the trace (what caused this missing), again for debugging purpose. 

====

Now those trace are nice for debugging, but the evaluator does not really care about those. 
So instead of having the evaluator to deal with rexOpMissing, it could be nice, before we send the plan to evaluator, does the conversion ourselves. 

```
// Plan for evaluator to consume
└── Query[tag=Plan-951622ec]
      └── Lit[value=MissingValueImpl(annotations=[]), tag=Plan-33b21fcd, static_type=missing]
```

Hence we have essentially two planner build for that purpose. 

We can put all those information to Rex.Op.MISSING in the public plan, but I having those as debugging node avoids backward incompatible changes.

Reply:
I can see how attaching this information is useful, but why is this **debug** in the public API plan? This makes sense for the internal plan, and we can dump these detailed messages during the PlanTransform.

I don't believe we should be increasing the public plan surface with debug information — error node is already a stretch..

Reply:
My comment is really asking, **could we please consolidate any error/debug messaging to the single err node**.

Reply:
The blocker here really comes from unresolved function: 

if you have `not_a_func(arg1, arg2)`

The internal plan is: 

To faithfully preserve the problem trace in the public plan as a PlanNode, we the have to wrap this unresolvedFunction in to a node. 

```
// What to turn that into the public node? 
internal data class Unresolved(
                @JvmField internal val identifier: Identifier,
                @JvmField internal val args: List<Rex>,
            )
```

The best option I see is to have a dedicated debug node: i.e., you have a debugging node for unresolved function and wrapper that node as the input for Rex.Op.Err. This way we fully preserved the node information and it is very easy to present to the customer about what exactly the error is.

I am sure we can hack around the issues, for example, we can construct some rex.op for all the args and left the function name in the error message, but this is a little hacky in my opinion.  

------- 

If we need the unresolved function node as a debug node, I don't see a reason why you should not have the missingOp and errOp as debug node. Separating those two in debug node makes debugging a little easier in my opinion.

Reply:
This is assumed that you want to use the plan not just the problem to debug: 

Take `not_a_function(atomic.no_exist_field)` for example 
Plan: 
```
--------Plan---------
⚬ PartiQLPlan[tag=Plan-9d5bede7]
   ├── Catalog[name=mode_test, tag=Plan-5b7090a4]
   │  └── Value[type=int2, tag=Plan-7097c826]
   └── Query[tag=Plan-4028b20a]
      └── Error[problem=-1:-1:<unknown>: Unknown function `not_a_function(<missing>), tag=Plan-70e40d7a, static_type=any]
         └── Unresolved[tag=Plan-2c91e120]
            ├── Symbol[caseSensitivity=INSENSITIVE, symbol=not_a_function, tag=Plan-2e518174]
            └── Missing[problem=-1:-1:<unknown>: Expression always returns null or missing: caused by Path Navigation failed - root is not a struct, tag=Plan-aa832a34, static_type=missing]
               └── Symbol[key=no_exist_field, tag=Plan-2ba2fcac]
                  └── Global[tag=Plan-ecd64321, static_type=int2]
                     └── Ref[catalog=0, symbol=0, tag=Plan-d4929e03]
```
Problem: 
```
-1:-1:<unknown>: Expression always returns null or missing: caused by Path Navigation failed - root is not a struct
-1:-1:<unknown>: Unknown function `not_a_function(<missing>)
```

In my opinion the plan is much easier.

Reply:
Each plan node has a unique `tag`. I suggest we return error nodes only, and any details are in a `Problem` which which is associated with the error node via the tag.

This would mean you don't need to stuff a bunch of debug stuff in the plan. And then we can expand upon the errors without breaking the plan API. Does this make sense?